lib/puma/request.rb

```diff
@@ -220,27 +220,33 @@ def prepare_response(status, headers, res_body, requests, client)
       cork_socket socket
 
       if resp_info[:no_body]
-        if content_length and status != 204
+        # 101 (Switching Protocols) doesn't return here or have content_length,
+        # it should be using `response_hijack`
+        unless status == 101
+          if content_length and status != 204
```

Comment:
let's just clean this up to `&&` while we're here (also on line 239)

Reply:
Done.